### DISCUSSION TITLE
/cmi maintenance update request

### DISCUSSION THREAD

**Discussion by mrfloris** [Zrips/CMI#7983] [Enhancement]
## Enhancements to CMI Maintenance mode. (greylisting)

- **Support for bungee/velocity** - so maintenance mode is only announced on one server, or on all servers. If `-network` or some param is used, trigger maintenance mode true/false on all the networked servers, not just the one (kinda like broadcast and bbroadcast)

- **Option to add a time(r)**, so players know in the server list when maintenance mode is ending approx. 

- **Option to show the remaining time counting down ingame** when a time is set, so players know how long maintenance might last.

- **Option to have a default `scheduler.yml` for maintenance mode**, that's false, but server owners can set to true, or run manually .. that explains to the players there's a maintenance mode in x minutes, lasting y time, and does or does Not end with a server restart.

- **A custom/alt server icon** when maintenance mode is set to true (and resets to default on false)

- **A custom/alt MOTD**, when maintenance mode is set to true (and resets to whatever it was before when set to false)

- **A bit more control over multi-line maintenance**, if we set something more complex now and use a bossbarmsg, the boss bar msg is HUGE and can't be read, but the 'you cannot join, maintenance mode is one, more info on discord, expecting to last x minutes, blah blah', would be nice to have as a 'default' configuration setting, so the `/cmi maintenance true "REASON"` becomes the dynamic part.

- **Move messages like those ^^ to a language file**, so localization is easier. 

- **have `/cmi reload` update the locale or config values for maintenance mode**, so we can live-edit in case the situation changes.

- **a quick command like `/cmi maintenance -add player`*, that triggers something like `/lp user player permission set cmi.maintenance.bypass true`, or whatever the perm node is.
`cmi.maintenance.add` (allows adding players to greylist)

- **`/cmi maintenance -image maintenance.png`** (plugins/cmi/images/maintenance.png) (set alt image when maintenance is set to true)
`cmi.maintenance.image` (allows setting alt server img)

- **`/cmi maintenance -motd "%cmi_maintenance_reason%"`** (set alt motd when maintenance is set to true)
`cmi.maintenance.motd` (allows setting alt motd)

- **`/cmi maintenance -status`** (shows current locale/config default values, the reason, and if it is set true/false, if alt motd, and alt img are set, displays any set timers, if scheduler is active, etc).
`cmi.maintenance.status` (allows requesting status)

**Comment by CloudeLecaw** [comment]
Please make this dream come true <3 XD

Im badly badly intested into the image part xD 
The motdpart we manged with creplace the regular cmi mantenance command with a custom alias that set the motd and maintenance in secound command. 

but at least if the command itself would have this option it also would be nice <3
